kubernetes/kubernetes.go

```diff
@@ -234,7 +256,32 @@ func isCustomS3UploadEnvsSet(ka *KubeAgentConfig) bool {
 	}
 	log.Infof("Detected custom S3 bucket location and S3 bucket region. "+
 		"Will upload collected metrics to %s in the aws region %s", ka.CustomS3UploadBucket, ka.CustomS3Region)
-	return true
+	return true, "aws"
+}
+
+// isCustomAzureUploadEnvsSet checks to see if the agent has custom Azure credentials needed for uploads
+// if these vars are found, set the env vars needed for the azure client. If not set, default upload to Apptio S3
+func isCustomAzureUploadEnvsSet(ka *KubeAgentConfig) (bool, string) {
+	if ka.CustomAzureUploadBlobContainerName == "" && ka.CustomAzureBlobURL == "" && ka.CustomAzureClientID == "" &&
+		ka.CustomAzureClientSecret == "" && ka.CustomAzureTenantID == "" {
+		if ka.APIKey == "" {
+			log.Fatalf("Invalid agent configuration. CLOUDABILITY_API_KEY is required " +
+				"when not using CLOUDABILITY_CUSTOM_AZURE_BLOB env vars.")
+		}
+		return false, ""
+	}
+	if ka.CustomAzureUploadBlobContainerName == "" || ka.CustomAzureBlobURL == "" || ka.CustomAzureClientID == "" ||
+		ka.CustomAzureClientSecret == "" || ka.CustomAzureTenantID == "" {
+		log.Fatalf("Invalid agent configuration. Detected only one of the six required environment variables "+
+			"to run in custom Azure upload mode. CLOUDABILITY_CUSTOM_AZURE_BLOB_CONTAINER_NAME is set to %s. "+
+			"CLOUDABILITY_CUSTOM_AZURE_BLOB_URL is set to %s CLOUDABILITY_CUSTOM_AZURE_TENANT_ID is set to %s. "+
+			"CLOUDABILITY_CUSTOM_AZURE_CLIENT_ID is set to %s. CLOUDABILITY_CUSTOM_AZURE_CLIENT_SECRET is set to %s.",
+			ka.CustomAzureUploadBlobContainerName, ka.CustomAzureBlobURL, ka.CustomAzureClientID, ka.CustomAzureTenantID,
+			ka.CustomAzureClientSecret)
```

Comment:
Don't log out secrets or other sensitive data. Can check that they != "" and log that they are set if you want context for them.